src/main/java/lotto/Lotto.java

```diff
@@ -0,0 +1,87 @@
+package lotto;
+
+import exception.CustomException;
+import lotto.exception.LottoErrorCode;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Lotto {
+
+    public static final int SELECT_SIZE = 6;
+
+    private final List<LottoNumber> numbers;
+
+    private Lotto(List<LottoNumber> numbers) {
+        checkOverlap(numbers);
+        this.numbers = numbers;
+    }
+
+    private void checkOverlap(List<LottoNumber> numbers) {
+        List<LottoNumber> distinct = numbers.stream()
+                .distinct()
+                .collect(Collectors.toList());
+
+        if (distinct.size() != numbers.size()) {
+            throw new CustomException(LottoErrorCode.LOTTO_NUMBER_OVERLAP);
+        }
+    }
+
+    public static Lotto generate() {
+        List<LottoNumber> lottoNumbersRange = LottoNumber.getLottoNumbersRange();
+        Collections.shuffle(lottoNumbersRange);
+        Lotto lotto = new Lotto(lottoNumbersRange.subList(0, SELECT_SIZE));
+        lotto.numbers.sort(Comparator.comparing(LottoNumber::getNumber));
+        return new Lotto(lottoNumbersRange.subList(0, SELECT_SIZE));
+    }
+
+    public static Lotto from(Lotto lotto) {
+        return new Lotto(lotto.numbers);
+    }
+
+    public static Lotto from(List<LottoNumber> lottoNumbers) {
+        checkLottoSize(lottoNumbers.size());
+        return new Lotto(lottoNumbers);
+    }
+
+    public static Lotto from(String stringLottoNumbers) {
+        String[] split = stringLottoNumbers.split(", ");
+        return from(LottoNumber.from(split));
+    }
+
+    private static void checkLottoSize(int size) {
+        if (size != SELECT_SIZE) {
+            throw new CustomException(LottoErrorCode.LOTTO_SIZE_BAD_REQUEST);
+        }
+    }
+
+    public List<LottoNumber> getLottoNumbers() {
+        List<LottoNumber> result = new ArrayList<>();
+        for (LottoNumber lotto : numbers) {
+            result.add(LottoNumber.from(lotto));
+        }
+        return result;
+    }
+
+    public Long containCount(Lotto lotto) {
+        return this.numbers.stream().filter(lotto.numbers::contains).count();
```

Comment:
한 라인에 온점(.)을 하나만 적는 것은 어떨까요?
```suggestion
        return this.numbers.stream()
            .filter(lotto.numbers::contains)
            .count();
```